core/lib/constants/src/ethereum.rs

```diff
@@ -8,7 +8,8 @@ pub static ETHEREUM_ADDRESS: Address = Address::zero();
 /// This the number of pubdata such that it should be always possible to publish
 /// from a single transaction. Note, that these pubdata bytes include only bytes that are
 /// to be published inside the body of transaction (i.e. excluding of factory deps).
-pub const GUARANTEED_PUBDATA_PER_L1_BATCH: u64 = 4000;
+pub const GUARANTEED_PUBDATA_PER_L1_BATCH_PRE_1_4_1: u64 = 4000;
+pub const GUARANTEED_PUBDATA_PER_L1_BATCH_1_4_1: u64 = 2500;
```

Comment:
Looking at all these changes (including the one from https://github.com/matter-labs/zksync-era/pull/809), I'm wondering if it would make more sense to have all these inside the zk_evm crate. In that case we'd just need to reference the new crate version.

@Deniallugo I'm sure you've already thought about it - what's the current stance? It will probably only be feasible when we integarte JayVM

Reply:
These changes are just not inherent to `zk_evm`. Things like `MAX_BOOTLOADER_MEMORY`/`MAX_TXS_PER_BATCH`/`GUARANTEED_PUBDATA`/etc are constants that are related to a specific bootloader version. I used the "1_4_1" here as it is how the upgrade will be called (I can call it some other name)

Reply:
I agree. I'm just missing some abstraction that would encapsulate everything related to a partoicular protocol version.
No need for any changes in this PR

Reply:
yeah, we could probably need an abstraction for bootloader and configs